test/Yoti.Auth.Tests/ShareUrl/Policy/DynamicPolicyBuilderTests.cs

```diff
@@ -0,0 +1,203 @@
+        {
+            WantedAttribute wantedAttribute = new WantedAttributeBuilder()
+                .WithName("SomeAttributeName")
+                .Build();
+
+            DynamicPolicy result = new DynamicPolicyBuilder()
+                .WithWantedAttribute(wantedAttribute)
+                .WithWantedAttribute(wantedAttribute)
+                .Build();
+
+            Assert.AreEqual(1, result.WantedAttributes.Count);
+            Assert.IsTrue(result.WantedAttributes.Contains(wantedAttribute));
+        }
+
+        [TestMethod]
+        public void BuildWithAttributes()
+        {
+            DynamicPolicy dynamicPolicy = new DynamicPolicyBuilder()
+                .WithFamilyName()
+                .WithGivenNames()
+                .WithFullName()
+                .WithDateOfBirth()
+                .WithGender()
+                .WithPostalAddress()
+                .WithStructuredPostalAddress()
+                .WithNationality()
+                .WithPhoneNumber()
+                .WithSelfie()
+                .WithEmail()
+                .WithAgeOver(55)
+                .WithAgeUnder(18)
+                .Build();
+
+            ICollection<WantedAttribute> result = dynamicPolicy.WantedAttributes;
+            var attributeMatcher = new WantedAttributeMatcher(result);
+
+            Assert.AreEqual(13, result.Count);
+
+            Assert.IsTrue(attributeMatcher.ContainsAttribute(UserProfile.FamilyNameAttribute, optional: false));
+            Assert.IsTrue(attributeMatcher.ContainsAttribute(UserProfile.GivenNamesAttribute, optional: false));
+            Assert.IsTrue(attributeMatcher.ContainsAttribute(UserProfile.FullNameAttribute, optional: false));
+            Assert.IsTrue(attributeMatcher.ContainsAttribute(UserProfile.GenderAttribute, optional: false));
+            Assert.IsTrue(attributeMatcher.ContainsAttribute(UserProfile.PostalAddressAttribute, optional: false));
+            Assert.IsTrue(attributeMatcher.ContainsAttribute(UserProfile.StructuredPostalAddressAttribute, optional: false));
+            Assert.IsTrue(attributeMatcher.ContainsAttribute(UserProfile.NationalityAttribute, optional: false));
+            Assert.IsTrue(attributeMatcher.ContainsAttribute(UserProfile.PhoneNumberAttribute, optional: false));
+            Assert.IsTrue(attributeMatcher.ContainsAttribute(UserProfile.SelfieAttribute, optional: false));
+            Assert.IsTrue(attributeMatcher.ContainsAttribute(UserProfile.EmailAddressAttribute, optional: false));
+            Assert.IsTrue(attributeMatcher.ContainsAttribute(UserProfile.DateOfBirthAttribute, optional: false));
+            Assert.IsTrue(attributeMatcher.ContainsAttribute(UserProfile.DateOfBirthAttribute, optional: false, derivation: $"{Constants.UserProfile.AgeOverAttribute}:55"));
+            Assert.IsTrue(attributeMatcher.ContainsAttribute(UserProfile.DateOfBirthAttribute, optional: false, derivation: $"{Constants.UserProfile.AgeUnderAttribute}:18"));
```

Comment:
_(Same as other comments)_ Perhaps we should also check the serialised JSON?

Reply:
Added a test for this in 91595f3